cmd/determinize-ci-operator/main.go

```diff
@@ -53,3 +92,52 @@ func main() {
 		}
 	}
 }
+
+func migrateOpenshiftInstallerSRCTemplates(
+	configuration *config.DataWithInfo,
+	allowedBranches sets.String,
+	allowedOrgs sets.String,
+	allowedCloudproviders sets.String,
+) (migratedCount int) {
+	if (len(allowedBranches) != 0 && !allowedBranches.Has(configuration.Info.Branch)) || (len(allowedOrgs) != 0 && !allowedOrgs.Has(configuration.Info.Org)) {
+		return 0
+	}
+
+	for idx, test := range configuration.Configuration.Tests {
+		if test.OpenshiftInstallerSrcClusterTestConfiguration == nil ||
+			(len(allowedCloudproviders) != 0 && !allowedCloudproviders.Has(string(test.OpenshiftInstallerSrcClusterTestConfiguration.ClusterProfile))) {
```

Comment:
are we going to run into an issue when we allow N templates but the full cross product with M cloud providers is not supported and we need allowed providers per template?

Reply:
We do have this issue already, because for example there is no `ipi-vsphere`. The purpose of this PR is to provide an automatic migration and flags to limit it so we won't be ending up doing `git add -p`, but actually doing the migration is a follow-up